src/FunctionValidationTrait.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+use LogicException;
+use ReflectionFunctionAbstract;
+use ReflectionParameter;
+
+trait FunctionValidationTrait
+{
+    abstract public function validateFirstParameter(ReflectionParameter $param): void;
+
+    abstract public function throwInvalidFunctionException(): void;
```

Comment:
nit: add throws typehint
```suggestion
    /**
     * @throws LogicException
     */
    abstract public function throwInvalidFunctionException(): void;
```